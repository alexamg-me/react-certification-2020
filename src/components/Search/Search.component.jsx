import React from 'react';
import { Wrapper, Icon, Input } from './Search.styled';

function Search({ query, setQuery }) {
  const onChangeHandler = (event) => {
    setQuery(event.target.value);
  };

  return (
    <Wrapper>
      <Icon />
      <Input type="text" value={query} onChange={onChangeHandler} />
    </Wrapper>
  );
}

export default Search;
