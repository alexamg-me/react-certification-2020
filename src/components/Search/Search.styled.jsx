import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  position: relative;
`;
const Input = styled.input`
  border: 1px solid #134d45;
  border-radius: 20px;
  height: 35px;
  min-width: 200px;
  padding: 10px;
  padding-right: 35px;
  font-size: 20px;
  color: #409287;
  background-color: #e2faf7;
  @media (max-width: 500px) {
    display: none;
  }
`;
const Button = styled.div`
  border-radius: 20px;
  height: 35px;
  width: 35px;
  cursor: pointer;
  transform: translateX(-33px) translateY(3px);
`;

export { Wrapper, Input, Button };
