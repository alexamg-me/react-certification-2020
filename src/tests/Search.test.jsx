import React from 'react';
import TestRenderer from 'react-test-renderer';
import Search from '../components/Search/Search.component';

describe(`Component: Search`, () => {
  test(`Search renders with default props`, () => {
    const component = TestRenderer.create(<Search />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
