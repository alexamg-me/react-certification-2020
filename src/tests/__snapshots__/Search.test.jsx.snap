// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Search Search renders with default props 1`] = `
<div
  className="sc-bdfBwQ ffLsOo"
>
  <div
    className="sc-gsTCUz HFjtR"
  />
  <input
    className="sc-dlfnbm ifQijI"
    type="text"
  />
</div>
`;
